lib/google/ads/google_ads/google_ads_client.rb

```diff
@@ -300,6 +301,14 @@ def lookup_util
         def default_api_version
           Google::Ads::GoogleAds.default_api_version
         end
+
+        def deprecate(deprecation)
+          @deprecation ||= Deprecation.new(
```

Comment:
Should we reinstantiate if config changed?

Reply:
I think it's unlikely that a user will change their deprecation configs during mid execution, WDYT?

Reply:
Also then it loses the state of where various warnings were created from. This should be fine.